### PR TITLE
[CI:DOCS] Implement PR template to assist review & release

### PR DESCRIPTION
**Pull request by cevich**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This duplicates the template used for buildah.  The intention
is to make it immediately clear to reviewers:

* The intended/basic purpose of the PR (also machine readable)
* Why are changes being proposed
* If there are any specific items need additional checking or scrutiny
* What should go into the release-notes (if anything).

#### How to verify it

You can't, it will only take effect on all new PRs after this one is merged.

#### Which issue(s) this PR fixes:

This was talked about and requested ages ago during a team meeting.  Recently I smelled it's ripened enough to be plucked off my TODO list :grin: 

#### Special notes for your reviewer:

You can see the rendered markdown, by click the **view file** link in the github WebUI under the **files changed** tab

#### Does this PR introduce a user-facing change?

None

### CODE REVIEW

**Review by ashley-cui — COMMENTED** [review]
Two small requests: Can we change the master -> main in the link to the contributing, and also remind the submitter to make sure that they've signed their commit using `git commit -s`? 

Other than that, LGTM!

**Comment by cevich** [comment]
Good catches.  Fixed.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,68 @@
...
+> /kind api-change
+> /kind bug
+> /kind cleanup
+> /kind deprecation
+> /kind design
+> /kind documentation
+> /kind failing-test
+> /kind feature
+> /kind flake
+> /kind other
```
Comment (edsantiago) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: Podman doesn't actually have most of these labels. And TBH, the ones we do have (bug, cleanup, documentation) we don't seem to really use.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,7 +1,68 @@
...
+Automatically closes linked issue when PR is merged.
+Uncomment the following comment block and include the issue
+number or None on one line.
+Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`, or `None`.
+-->
+
+<!--
+Fixes #
+or
+None
```
Comment (edsantiago) [inline .github/PULL_REQUEST_TEMPLATE.md:49]: Has anyone ever done this `None` thing? If so why? It just seems like clutter; I vote to just say "Only include Fixes if the PR really fixes something"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mheon) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: Concur, this isn't really necessary IMO. Tagging issues is sufficient, we don't need to tag PRs.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mheon) [reply .github/PULL_REQUEST_TEMPLATE.md:49]: Also, please make it clear that `Fixes #` should also be in the commit message, for my sanity when writing release notes.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,7 +1,68 @@
...
+requires additional action from users switching to the new release, include the
+string "action required".
+
+For more information on release notes please follow the kubernetes model:
+https://git.k8s.io/community/contributors/guide/release-notes.md
+-->
+
+```release-note
+
+```
```
Comment (TomSweeneyRedHat) [inline .github/PULL_REQUEST_TEMPLATE.md:68]: Is the lack of a release note entry something we could fail in the smoke test?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: My understanding (could be wrong) is the bot will create and maintain the labels for us.  Whatever we write in the template is what will be used.  I think the purpose is to help at release-notes time, so you can filter out all the PRs having a docs or flakes etc.  In any case, I'm in agreement we should use the `/kind` list (and labels) that are most relevant/helpful.  Someone just needs to tell me what that set is, and I'll make it happen.

Note: Github treates PRs and Issues as the same kind of "thing".  The same set of labels can be applied to both, there's no such thing as PR or issue-only labels.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:49]: I can, but...this is easily missed/forgotten, even if it's in the template.  I think that's what the labels are suppose to be used for (filtering).

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:68]: Yes in fact we can!  The entire PR Description is available as an env. var for us to do what we please with.  The challenge is in coming up with the right script to do it :grin:

Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:68]: Clarification: I'm willing to take on this work...after this PR is finalized and merged or closed.

**Comment by cevich** [comment]
Updated: Removed the entire `/kind blah` section.  Reworded most of the comment blocks.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,7 +1,55 @@
...
+
+<!---
+Please specify the precise conditions and/or the specific test(s) which must pass.
+-->
+
+#### Which issue(s) this PR fixes:
+
+<!--
+Please uncomment this block and include only one of the following on a
+line by itself:
```
Comment (mheon) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: Please also mention to include this in the commit message.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,7 +1,55 @@
...
+
+Fixes #<issue number>
+
+*** Please also put 'Fixes #' in the PR title ***
+
+-->
+
+#### Special notes for your reviewer:
+
+#### Does this PR introduce a user-facing change?
```
Comment (mheon) [inline .github/PULL_REQUEST_TEMPLATE.md:41]: This is of somewhat limited utility because my existing release notes tooling works based on commit messages, not PRs (by necessity - our backports process makes identifying what commits are in a release, that were not in a previous release, quite complicated). I could theoretically work backwords from that to find the PR and get the release notes, but it would get time consuming (and I don't think it's easily scriptable)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: You know actually...This might be another item like Tom suggested, we can check for it early on with CI:

If any of the commits, PR title, or this section mention 'Fixes #whatever`: Immediately fail the build if that same thing isn't present in the other expected places.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:41]: Okay, so we should just get rid of this section too then?

**Comment by cevich** [comment]
Update: Include `Fixes #` in commit description, and removed release-notes section.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (mheon) [reply .github/PULL_REQUEST_TEMPLATE.md]: Either that or tell them to add it directly to `RELEASE_NOTES.md`

Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md]: Oh! That's a novel idea, I like it!  But how could it be enforced?.  Humans would probably just ignore it in a template-comment :disappointed: 

I could see doing it by adding back in a `/kind` tagging mechanism.  Cirrus-CI supports running certain tests based on the PR labels.  For example, if a `kind:userfacing` tag is present, CI can make sure the`RELEASE_NOTES.md` file changed.  That way the requirement has "teeth".

**Comment by cevich** [comment]
I think this is ready to be merged or closed (no hard feelings either way).  If merged, I'll need to add a jira-card re: adding some checks into CI to verify content.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,7 +1,39 @@
...
+Please uncomment this block and include only one of the following on a
+line by itself:
+
+None
+
+-OR-
+
+Fixes #<issue number>
+
+*** Please also put 'Fixes #' in the PR title and commit description***
```
Comment (Luap99) [inline .github/PULL_REQUEST_TEMPLATE.md:35]: Fixes should not be part of the PR title

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,7 +1,39 @@
 <!--
 Thanks for sending a pull request!
 
-Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/master/CONTRIBUTING.md#submitting-pull-requests).
+Please make sure you've read our contributing guidelines and how to submit a pull request (https://github.com/containers/podman/blob/main/CONTRIBUTING.md#submitting-pull-requests).
 
 In case you're only changing docs, make sure to prefix the pull-request title with "[CI:DOCS]".  That will prevent functional tests from running and save time and energy.
+
+Finally, be sure to sign your commits (e.g. `git commit -s`)
```
Comment (Luap99) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: 1) with your real name, no nick names
2) If they try to open a PR and read the template, they already have a commit, so I guess we should say `git commit -s --amend` and force push.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:35]: @mheon wanted this.

Comment (Luap99) [reply .github/PULL_REQUEST_TEMPLATE.md:35]: It should be part of the PR description but not the title.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md]: Okay, fixed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: I'm not opposed to whatever, I'm just pretty sure Matt wanted it in the title and description, but maybe I'm wrong. His comment is long since lost to `--force` push history :confused:

Comment (mheon) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: Description is better than title, sorry if I was unclear.

Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: @mheon when you have a moment, could you clarify what/where you'd like the `FIXES #<whatever>` text to appear?

Comment (mheon) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: PR description and commit description both. Can be in the title, but does not need to be.

Comment (cevich) [reply .github/PULL_REQUEST_TEMPLATE.md:37]: Got it, thanks for clarification.
